ucfopen/Materia
# Issue/894 s3 upload

### Testing New S3 Upload Features Locally

**These branches must be functioning, and finished before merging this request:**
- [ ] https://github.com/ucfcdl/materia-client-assets/pull/11
- [ ] https://clu.cdl.ucf.edu/materia/materia-docker/merge_requests/12
- [ ] https://clu.cdl.ucf.edu/serverless/materia-thumbnail-generator/merge_requests/3
- [x] FakeS3 source branch

**Notes:** 
- For best results, test in Incognito Mode, or any form of private browsing to avoid caching.
- Before assuming an image is not displaying due to a bug, refresh the browser page.
- When switching between s3 and plupload, be sure to truncate the asset table in the materia database to avoid confusion.
 
#### Fresh Install of Materia with S3 Upload Feature
1. Clone the *materia-docker* repository from Clu
2. Switch to branch *issue/876-add-fake-s3-container-to-materia-docker* in the *materia-docker* repo, and run *run_first.sh*
3. Go to the *app* directory of *materia-docker*, and switch to branch *issue/894-s3-upload*.
4. Run migrations :`docker-compose run --rm phpfpm php oil r migrate`
5. Go to the *materia-thumbnail-generator* directory of *materia-docker*, and switch to branch *condense-repo*
6. Yarn Link Materia's *materia-client-assets* dependency.
     - Clone the *materia-client-assets* repository from GitHub anywhere outside of the *materia-docker* repository
     - Switch to branch *issue/894-s3-upload* in the *materia-client-assets* repo
     - Run the command `yarn link` in the *materia-client-assets* repo
     - Switch to the *app* directory of the *materia-docker* repo, and run the command `yarn watch`
     - In a new terminal window, run the command `yarn link "materia-client-assets"` in the *app* directory of *materia-docker*
7. Install a few widgets (through devmateria) that support media uploads

#### Setup Materia to Use Plupload (current upload process)
1. Run the command `docker-compose stop` to stop running containers
2. Go to *fuel/app/config/development/materia.php* and set `s3_enabled: false`
3. Run the `docker-compose up` machine to start containers.
4. Refresh any previously opened pages

#### Setup Materia to Use Local S3 (new feature)
1. Run the command `docker-compose stop` to stop running containers
2. Go to *fuel/app/config/development/materia.php* and follow the comments to set the right s3_config values.
3. Run the `docker-compose up` machine to start containers.
4. Refresh any previously opened pages

#### Suggested Actions to Test for S3 AND Plupload
- uploaded images appear in widget as expected
- image thumbnail displays in the media uploader as expected
- widget preview displays images correctly
- published widget displays images correctly
- clicking an images thumbnail in the media uploader adds the correct image to the widget
- editing images on a published images correctly updates the published widget
only jpg, jpeg and png can be uploaded through the media uploader
- upload mp3 files and make sure they play in all situations (published, creator, preview)
- upload wav files and make sure they play in all situations (published, creator, preview)
- **try to break it with actions not on this list**

### **Testing New S3 Upload Features Remotely**
To Be Continued...
